Source/Scene/ModelExperimental/InstancingPipelineStage.js

```diff
@@ -0,0 +1,284 @@
+  var instancingTranslationMin = new Cartesian3(
+    Number.MAX_VALUE,
+    Number.MAX_VALUE,
+    Number.MAX_VALUE
+  );
+
+  // Translations get initialized to (0, 0, 0).
+  var translationTypedArray = defined(translationAttribute)
+    ? translationAttribute.typedArray
+    : new Float32Array(count * 3);
+  // Rotations get initialized to (0, 0, 0, 0). The w-component is set to 1 in the loop below.
+  var rotationTypedArray = defined(rotationAttribute)
+    ? rotationAttribute.typedArray
+    : new Float32Array(count * 3);
+  // Scales get initialized to (1, 1, 1).
+  var scaleTypedArray = defined(scaleAttribute)
+    ? scaleAttribute.typedArray
+    : new Float32Array(count * 3);
+  scaleTypedArray.fill(1);
+
+  var setRotationW = defined(rotationAttribute);
+
+  for (var i = 0; i < count; i++) {
+    var translation = new Cartesian3(
+      translationTypedArray[i * 3],
+      translationTypedArray[i * 3 + 1],
+      translationTypedArray[i * 3 + 2]
+    );
+
+    Cartesian3.maximumByComponent(
+      instancingTranslationMax,
+      translation,
+      instancingTranslationMax
+    );
+    Cartesian3.minimumByComponent(
+      instancingTranslationMin,
+      translation,
+      instancingTranslationMin
+    );
+
+    var rotation = new Quaternion(
+      rotationTypedArray[i * 4],
+      rotationTypedArray[i * 4 + 1],
+      rotationTypedArray[i * 4 + 2],
+      setRotationW ? rotationTypedArray[i * 4 + 3] : 1
+    );
+
+    var scale = new Cartesian3(
+      scaleTypedArray[i * 3],
+      scaleTypedArray[i * 3 + 1],
+      scaleTypedArray[i * 3 + 2]
+    );
+
+    var transform = Matrix4.fromTranslationQuaternionRotationScale(
+      translation,
+      rotation,
+      scale,
+      transformScratch
+    );
+
+    var offset = elements * i;
+
+    transformsTypedArray[offset + 0] = transform[0];
+    transformsTypedArray[offset + 1] = transform[4];
+    transformsTypedArray[offset + 2] = transform[8];
+    transformsTypedArray[offset + 3] = transform[12];
+    transformsTypedArray[offset + 4] = transform[1];
+    transformsTypedArray[offset + 5] = transform[5];
+    transformsTypedArray[offset + 6] = transform[9];
+    transformsTypedArray[offset + 7] = transform[13];
+    transformsTypedArray[offset + 8] = transform[2];
+    transformsTypedArray[offset + 9] = transform[6];
+    transformsTypedArray[offset + 10] = transform[10];
+    transformsTypedArray[offset + 11] = transform[14];
+
+    renderResources.instancingTranslationMax = instancingTranslationMax;
+    renderResources.instancingTranslationMin = instancingTranslationMin;
+  }
+
+  return transformsTypedArray;
+}
+
+function processMatrixAttributes(node, renderResources, frameState) {
+  var transformsTypedArray = getInstanceTransformsTypedArray(
+    node.instances,
+    renderResources
+  );
+  var transformsVertexBuffer = Buffer.createVertexBuffer({
+    context: frameState.context,
+    typedArray: transformsTypedArray.buffer,
+    usage: BufferUsage.STATIC_DRAW,
+  });
```

Comment:
Need to set `transformsVertexBuffer.vertexArrayDestroyable = false` otherwise it will get destroyed twice, once when the vertex array is destroyed, and again when the model `_resources` are destroyed.